sklearn/isotonic.py

```diff
@@ -271,6 +272,7 @@ def _build_y(self, X, y, sample_weight, trim_duplicates=True):
         check_consistent_length(X, y, sample_weight)
         X, y = [check_array(x, ensure_2d=False) for x in [X, y]]
 
+        X = as_float_array(X)
```

Comment:
This will trigger a copy, and is rather redundant with `check_array`.
I would prefer:
```py
X = check_array(X, dtype=[np.float64, np.float32])
y = check_array(y, dtype=X.dtype, ensure_2d=False)
```

Reply:
Failing tests on travis seem to suggest that my assumptions here fail on some configs; in particular this:

```
input dtype	 int32
as_float_array	 float32
check_array	 float64
```